bundles/org.openhab.binding.broadlink/src/main/java/org/openhab/binding/broadlink/handler/BroadlinkBaseThingHandler.java

```diff
@@ -0,0 +1,293 @@
+
+    public final Logger logger = LoggerFactory.getLogger(getClass());
+    private static final String INITIAL_DEVICE_ID = "00000000";
+
+    protected BroadlinkDeviceConfiguration thingConfig = new BroadlinkDeviceConfiguration();
+
+    private @Nullable RetryableSocket socket;
+
+    // Can be injected for test purposes
+    private @Nullable NetworkTrafficObserver networkTrafficObserver;
+
+    private int count;
+    private @Nullable ScheduledFuture<?> refreshHandle;
+    private boolean authenticated = false;
+    // These get handed to us by the device after successful authentication:
+    private byte[] deviceId;
+    private byte[] deviceKey;
+
+    public BroadlinkBaseThingHandler(Thing thing) {
+        super(thing);
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+    }
+
+    // For test purposes
+    void setSocket(RetryableSocket socket) {
+        this.socket = socket;
+    }
+
+    void setNetworkTrafficObserver(NetworkTrafficObserver networkTrafficObserver) {
+        this.networkTrafficObserver = networkTrafficObserver;
+    }
+
+    private boolean hasAuthenticated() {
+        return authenticated;
+    }
+
+    @Override
+    public void initialize() {
+        this.thingConfig = getConfigAs(BroadlinkDeviceConfiguration.class);
+        count = (new Random()).nextInt(65535);
+
+        if (this.socket == null) {
+            this.socket = new RetryableSocket(thingConfig, logger);
+        }
+
+        updateItemStatus();
+
+        if (thingConfig.getPollingInterval() != 0) {
+            refreshHandle = scheduler.scheduleWithFixedDelay(new Runnable() {
+
+                @Override
+                public void run() {
+                    updateItemStatus();
+                }
+            }, 1L, thingConfig.getPollingInterval(), TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    @SuppressWarnings("null")
+    public void dispose() {
+        if (refreshHandle != null && !refreshHandle.isDone()) {
+            boolean cancelled = refreshHandle.cancel(true);
+            logger.debug("Cancellation successful: {}", cancelled);
+        }
+        if (socket != null) {
+            socket.close();
+            socket = null;
+        }
+        super.dispose();
+    }
+
+    private boolean authenticate() {
+        authenticated = false;
+        // When authenticating, we must ALWAYS use the initial values
+        this.deviceId = HexUtils.hexToBytes(INITIAL_DEVICE_ID);
+        this.deviceKey = HexUtils.hexToBytes(BroadlinkBindingConstants.BROADLINK_AUTH_KEY);
+
+        try {
+            byte authRequest[] = buildMessage((byte) 0x65, BroadlinkProtocol.buildAuthenticationPayload(), -1);
+            byte response[] = sendAndReceiveDatagram(authRequest, "authentication");
+            if (response == null) {
+                logger.warn("response from device during authentication was null");
```

Comment:
Is it possible to give a warning that guides the user what they need to change? I have not read all code so if this is provided by the thing going offline with a helpful message then that is enough.

Reply:
Added actionable guidance text to check the MAC address of the device is configured correctly.